dashboard/test/controllers/maker_controller_test.rb

```diff
@@ -63,117 +37,19 @@ class MakerControllerTest < ActionController::TestCase
-    create :user_script, user: @student, script: @csd6_2018, assigned_at: Time.now
-    refute_includes @student.scripts, @csd6_2017
-    assert_includes @student.scripts, @csd6_2018
-
-    assert_equal @csd6_2018, MakerController.maker_script(@student)
-  end
-
-  test "shows CSD6-2017 if CSD6-2017 is assigned" do
-    create :user_script, user: @student, script: @csd6_2017, assigned_at: Time.now
-    assert_includes @student.scripts, @csd6_2017
-    refute_includes @student.scripts, @csd6_2018
-
-    assert_equal @csd6_2017, MakerController.maker_script(@student)
-  end
-
-  test "shows CSD6-2018 if both CSD6-2017 and CSD6-2018 are assigned" do
-    create :user_script, user: @student, script: @csd6_2017, assigned_at: Time.now
-    create :user_script, user: @student, script: @csd6_2018, assigned_at: Time.now
-    assert_includes @student.scripts, @csd6_2017
-    assert_includes @student.scripts, @csd6_2018
-
-    assert_equal @csd6_2018, MakerController.maker_script(@student)
-  end
-
-  test "shows CSD6-2019 if both CSD6-2018 and CSD6-2019 are assigned" do
-    create :user_script, user: @student, script: @csd6_2018, assigned_at: Time.now
-    create :user_script, user: @student, script: @csd6_2019, assigned_at: Time.now
-    assert_includes @student.scripts, @csd6_2018
-    assert_includes @student.scripts, @csd6_2019
-
-    assert_equal @csd6_2019, MakerController.maker_script(@student)
-  end
-
-  test "shows CSD6-2019 if CSD-2019 is assigned" do
-    create :follower, section: create(:section, unit_group: @csd_2019), student_user: @student
-    assert_includes @student.section_courses, @csd_2019
-
-    assert_equal @csd6_2019, MakerController.maker_script(@student)
-  end
-
-  test "shows CSD6-2018 if CSD-2018 is assigned" do
-    create :follower, section: create(:section, unit_group: @csd_2018), student_user: @student
-    refute_includes @student.section_courses, @csd_2017
-    assert_includes @student.section_courses, @csd_2018
-
-    assert_equal @csd6_2018, MakerController.maker_script(@student)
-  end
-
-  test "shows CSD6-2017 if CSD-2017 is assigned" do
-    create :follower, section: create(:section, unit_group: @csd_2017), student_user: @student
-    assert_includes @student.section_courses, @csd_2017
-    refute_includes @student.section_courses, @csd_2018
-
-    assert_equal @csd6_2017, MakerController.maker_script(@student)
-  end
-
-  test "shows CSD6-2018 if both CSD-2017 and CSD-2018 are assigned" do
-    create :follower, section: create(:section, unit_group: @csd_2017), student_user: @student
-    create :follower, section: create(:section, unit_group: @csd_2018), student_user: @student
-    @student.reload
-    assert_includes @student.section_courses, @csd_2017
-    assert_includes @student.section_courses, @csd_2018
-
-    assert_equal @csd6_2018, MakerController.maker_script(@student)
-  end
-
-  test "shows CSD6-2018 if both CSD6-2017 and CSD-2018 are assigned" do
```

Comment:
These kinds of tests can be helpful for documenting desired behavior –– thoughts about doing something similar in a future PR, given that this functionality is already in our code?

Reply:
I haven't verified this locally, but it looks like the functionality in this particular test case was taken away by this PR. is the request to move this test over to cover the new implementation, e.g. replace "csd6" with "devices" below? If so, that makes sense to me. although, depending on how fully we move onto existing helper methods like `Unit.latest_stable_version`, the amount of dedicated test coverage this code path needs may be reduced.

Reply:
This is a good point, I think I'm moving fully to helper methods (including writing a new helper method in `unit.rb` and adding tests in `unit_test.rb`) so I'll add a note about this in [the upcoming cleanup PR](https://github.com/code-dot-org/code-dot-org/pull/49815) to get thoughts on whether this would be necessary!

Reply:
Sounds good to me. Agreed that if you're using helper methods that are tested, no need to include extra unit tests here 👍